src/cmd/flux-mini.py

```diff
@@ -395,46 +395,38 @@ def read_script(args):
                 return "#!/bin/sh\n" + " ".join(args.SCRIPT) + "\n"
 
             # O/w, open script for reading
-            name = args.SCRIPT[0]
-            filep = open(name, "r", encoding="utf-8")
+            name = open_arg = args.SCRIPT[0]
         else:
             name = "stdin"
-            filep = open(0, "r", encoding="utf-8")
-        try:
-            #  Read first two bytes and ensure it starts with #!
-            script = "#!/bin/sh\n" if args.wrap else filep.read(2)
-            if not args.wrap and script[:2] != "#!":
-                raise ValueError(f"{name} does not appear to start with '#!'")
-            #  Read rest of script
-            script += filep.read()
-        except UnicodeError:
-            raise ValueError(
-                f"{name} does not appear to be a script, or failed to encode as utf-8"
-            )
-        return script
+            open_arg = 0  # when passed to `open`, 0 gives the `stdin` stream
+        with open(open_arg, "r", encoding="utf-8") as filep:
+            try:
+                #  Read script
+                script = filep.read()
+                if args.wrap:
+                    script = "#!/bin/sh\n" + script
+            except UnicodeError:
+                raise ValueError(
+                    f"{name} does not appear to be a script, "
+                    "or failed to encode as utf-8"
+                )
+            return script
 
     def init_jobspec(self, args):
         # If no script (reading from stdin), then use "flux" as arg[0]
-        command = args.SCRIPT
-        if not command:
-            command = ["flux"]
-
         if not args.nslots:
             raise ValueError("Number of slots to allocate must be specified")
 
-        jobspec = JobspecV1.from_command(
-            command=command,
-            num_tasks=args.nslots,
-            cores_per_task=args.cores_per_slot,
-            gpus_per_task=args.gpus_per_slot,
+        jobspec = JobspecV1.from_batch_command(
+            script=self.read_script(args),
+            jobname=args.SCRIPT[0] if args.SCRIPT else "stdin",
```

Comment:
This changes the default argv0 for batch scripts given on stdin from "flux" to "stdin". Though I'm not sure it is a bad thing, I just wanted to point it out. To be honest, it is unlikely that reading a script from stdin will be all that common, and the job-name can be overridden on the command line. (And I do notice the new error in `from_batch_command()` requires that the job argv0 be "stdin" since it is used in the error message)

Reply:
Yeah, I made that change because I thought it was worthwhile for `from_batch_command` to check whether the script starts with `"#!"`, and I wanted to avoid a double-check. But if the argv0 is primarily used for displaying the jobs to a user, I think "stdin" isn't much better or worse than "flux".

Reply:
What about making argv0 default to something like `script[:50]`?

Reply:
The main point I was making is that this was an interface change, not called out in any commit message title or description. I also think stdin is no worse or better than "flux" (though "flux" is the *real* argv0, which is why that was the initial choice)

> I suspect that if I wasn't used to the whole nesting instances thing, and I saw "flux" on my jobs printout, I would think that it was just some system job---not one of mine

In the default `flux jobs` output, user name and job name are separate fields, so I don't think that there will be any confusion about whose jobs are whose.

> What about making argv0 default to something like script[:50]? Or "Nested Flux instance from stdin".

Job name should be fairly short, there is only 10 characters reserved for default output.

> And as someone who is a little used to nesting instances, I would probably want to know something about the initial program of the nested instance.

Agreed. As I said, the job name can always be set by user via `--job-name` option.  I think in this rare case where job script comes from stdin and job-name is not set, "stdin" seems fine.

I notice that the approach `sbatch` takes in this scenario is to name the job "sbatch" -- another option would be to default arg[0] to `batchscript` or something. At least that would make sense as an argument.

Reply:
> The main point I was making is that this was an interface change, not called out in any commit message title or description.

Oh, that makes a lot of sense. I'll fix that.

> I notice that the approach `sbatch` takes in this scenario is to name the job "sbatch" -- another option would be to default arg[0] to `batchscript` or something. At least that would make sense as an argument.

I quite like that, actually! I think `flux-mini: ERROR: batchscript does not appear to start with '#!'` would be a pretty reasonable error message. And as a jobname I think it's better. Anyone else want to chime in?

Reply:
I agree with all your points. Let's just go with batchscript if you are fine with it. Unfortunately, it is one character too long for default `flux jobs` output, but that hardly matters much.
```
ƒ(s=4,d=0,builddir) grondo@asp:~/git/f.git$ flux jobs -a
             JOBID USER     NAME       STATUS NTASKS NNODES  RUNTIME RANKS
   102791284523008 grondo   batchscrip     CD      1      1   2.444s 0
```

Reply:
Sounds good. I'll fix that, mark the change in the commit message, change the docstring you mentioned below, and force-push it.